gitrevise/odb.py

```diff
@@ -31,6 +31,8 @@
 if TYPE_CHECKING:
     from subprocess import _FILE
 
+    from typing_extensions import Self
```

Comment:
I don't love needing to pull in a dependency for `Self` when we can do the existing `TypeVar` stuff and it looks like it should work OK? What does this do differently?

Reply:
Depends how recent you wanna be: it's included in python proper as of py3.11: https://peps.python.org/pep-0673/

```python
from typing import Self
```

Reply:
Oh that's nice.

Looks like I last increased the minimum python version in 2022 from 3.6 to 3.8 (https://github.com/mystor/git-revise/commit/4a7e852ccb2a7319c1cd28fb2d360455682debed). I imagine we can get away with bumping the minimum version again so long as the major LTS distros (debian stable / ubuntu) are shipping that version of Python or newer.

Reply:
[debian stable (bookworm) is 3.11](https://wiki.debian.org/Python)
[Ubuntu 22.04 is on 3.10 and 24.04 on 3.12](https://packages.ubuntu.com/search?keywords=python3&searchon=names&exact=1&suite=all&section=all)

Not sure how dependent you are on system python for installing a newer version of any pyhton package though: if you can install a newer python package you probably alos can instll (and probably already are installing) a newer python :-)

Reply:
Cool. I'll admit I'm out-of-the-loop on these modern python packaging changes, and so I worry about things like "can someone on the Ubuntu LTS do `pip3 install --user git-revise` and have it work", but I suppose that's less of a thing folks are doing nowadays?

I think I'd definitely be OK with bumping the python version to 3.10, less certain about 3.11 given 22.04 is still on 3.10, but perhaps that is also OK.